src/UsgsAstroFramePlugin.cpp

```diff
@@ -277,207 +273,199 @@ csm::Model *UsgsAstroFramePlugin::constructModelFromISD(const csm::Isd &imageSup
-  }
-  else if (imageSupportData.param("itrans_sample", 2) == "") {
-    missingKeywords.push_back("itrans_sample missing third element");
-  }
-
-  sensorModel->m_ephemerisTime = atof(imageSupportData.param("ephemeris_time").c_str());
-  if (imageSupportData.param("ephemeris_time") == "") {
-    missingKeywords.push_back("ephemeris_time");
-  }
-
-  sensorModel->m_originalHalfSamples =
-      atof(imageSupportData.param("original_half_samples").c_str());
-
-  sensorModel->m_boresight[0] = atof(imageSupportData.param("boresight", 0).c_str());
-  sensorModel->m_boresight[1] = atof(imageSupportData.param("boresight", 1).c_str());
-  sensorModel->m_boresight[2] = atof(imageSupportData.param("boresight", 2).c_str());
-
-  sensorModel->m_iTransL[0] = atof(imageSupportData.param("itrans_line", 0).c_str());
-  sensorModel->m_iTransL[1] = atof(imageSupportData.param("itrans_line", 1).c_str());
-  sensorModel->m_iTransL[2] = atof(imageSupportData.param("itrans_line", 2).c_str());
-  if (imageSupportData.param("itrans_line", 0) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-  else if (imageSupportData.param("itrans_line", 1) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-  else if (imageSupportData.param("itrans_line", 2) == "") {
-    missingKeywords.push_back("itrans_line needs 3 elements");
-  }
-
-  sensorModel->m_nLines = atoi(imageSupportData.param("nlines").c_str());
-  sensorModel->m_nSamples = atoi(imageSupportData.param("nsamples").c_str());
-  if (imageSupportData.param("nlines") == "") {
-    missingKeywords.push_back("nlines");
-  }
-  if (imageSupportData.param("nsamples") == "") {
-    missingKeywords.push_back("nsamples");
-  }
-
-  sensorModel->m_transY[0] = atof(imageSupportData.param("transy", 0).c_str());
-  sensorModel->m_transY[1] = atof(imageSupportData.param("transy", 1).c_str());
-  sensorModel->m_transY[2] = atof(imageSupportData.param("transy", 2).c_str());
-  if (imageSupportData.param("transy", 0) == "") {
-    missingKeywords.push_back("transy 0");
-  }
-  else if (imageSupportData.param("transy", 1) == "") {
-    missingKeywords.push_back("transy 1");
-  }
-  else if (imageSupportData.param("transy", 2) == "") {
-    missingKeywords.push_back("transy 2");
-  }
-
-  sensorModel->m_transX[0] = atof(imageSupportData.param("transx", 0).c_str());
-  sensorModel->m_transX[1] = atof(imageSupportData.param("transx", 1).c_str());
-  sensorModel->m_transX[2] = atof(imageSupportData.param("transx", 2).c_str());
-  if (imageSupportData.param("transx", 0) == "") {
-    missingKeywords.push_back("transx 0");
-  }
-  else if (imageSupportData.param("transx", 1) == "") {
-    missingKeywords.push_back("transx 1");
-  }
-  else if (imageSupportData.param("transx", 2) == "") {
-    missingKeywords.push_back("transx");
+  if (imageSupportData.param("sensor_orientation", 3) == "") {
+    missingKeywords.push_back("kappa");
```

Comment:
The missing keywords for Omega, Phi, Kappa should be changed to just check if sensor orientation has a quaternion, 4 elements. If it doesn't, it should append "sensor_orientation" to the missingKeywords.